frame/identity/src/types.rs

```diff
@@ -93,6 +94,55 @@ impl Encode for Data {
 }
 impl codec::EncodeLike for Data {}
 
+/// Add a Raw variant with the given index and a fixed sized byte array
+macro_rules! data_raw_variants {
+    ($variants:ident, $(($index:literal, $size:literal)),* ) => {
+		$variants
+		$(
+			.variant(stringify!(Raw$size), |v| v
+				.index($index)
+				.fields(Fields::unnamed().field(|f| f.ty::<[u8; $size]>()))
+			)
+		)*
+    }
+}
+
+impl TypeInfo for Data {
+	type Identity = Self;
+
+	fn type_info() -> Type {
+		let variants = Variants::new()
+			.variant("None", |v| v.index(0));
+
+		// create a variant for all sizes of Raw data from 0-32
+		let variants = data_raw_variants!(variants,
+			(1, 0),   (2, 1),   (3, 2),   (4, 3),   (5, 4),   (6, 5),   (7, 6),   (8, 7),
+			(9, 8),   (10, 9),  (11, 10), (12, 11), (13, 12), (14, 13), (15, 14), (16, 15),
+			(17, 16), (18, 17), (19, 18), (20, 19), (21, 20), (22, 21), (23, 22), (24, 23),
+			(25, 24), (26, 25), (27, 26), (28, 27), (29, 28), (30, 29), (31, 30), (32, 31), (33, 32)
+		);
```

Comment:
This expands the `Raw(BoundedVec)` variant to a variant for each of the available sizes in the range `0..=32` with type `[u8; N]` e.g. `Raw0([u8; 0])`..`Raw32([u8; 32])`, See custom `Encode` impl above for how it is encoded.

Reply:
It's a ~~hack~~ workaround in order to generate a valid type description that will be able to decode the custom scheme here: https://github.com/paritytech/substrate/blob/9cbdff5e6be27e77b178737ac8fbaf56b0d3f97a/frame/identity/src/types.rs#L61-L66

The philosophy I have gone with here and for other types with custom non-derived encoding is to generate a valid type definition that will allow best-effort decoding without requiring any additional logic. This way given `Metadata` and a SCALE encoded `Vec<u8>` you can decode the whole thing without providing any additional information. 

In some cases there will be opaque data or a funky workaround as in this case, which would require some further post-processing or conversion to use properly. In a static context (generating types downstream) if we wanted to generate this type we would provide some override for it. But in a dynamic environment where some types just need to be skipped (e.g. when decoding a stream of `Events`) then this is useful because the metadata has enough information to decode.

Probably we can come up with a better approach with how to represent types with non-standard encoding, but I think this a good compromise for now.